impress-org/givewp
# Feature: Offline donations gateway v3 compatibility

## Description

This PR is a rebuild of the Offline gateway to make use of the Gateway API and be v2 + v3 compatible. It cleans up a bunch of the old code and should port everything over nicely. The ID and everything should be preserved such that it's full backwards-compatible.

Note, since there's now a Billing Address block, the billing address settings don't do anything anymore on v3 forms. It should still work on v2 forms, though.


## Affects

The offline donations gateway

## Visuals

### In the donation form
<img width="663" alt="image" src="https://github.com/impress-org/givewp/assets/2024145/b99e26b4-50ea-4b02-b051-0acf346f562c">

### Settings in form builder
<img width="723" alt="image" src="https://github.com/impress-org/givewp/assets/2024145/4a6b72d2-129a-44e7-834e-4c18837f55b8">



## Testing Instructions

Make sure the Offline gateway works with v2 and v3 forms. Make sure to test out the various settings and emails related to offline.

## Pre-review Checklist
